components/brave_wallet/browser/zcash/zcash_create_orchard_to_transparent_transaction_task.cc

```diff
@@ -0,0 +1,183 @@
+    return;
+  }
+
+  if (!spendable_notes_) {
+    GetSpendableNotes();
+    return;
+  }
+
+  if (!transaction_) {
+    CreateTransaction();
+    return;
+  }
+
+  std::move(callback_).Run(std::move(transaction_.value()));
+}
+
+void ZCashCreateOrchardToTransparentTransactionTask::GetSpendableNotes() {
+  if (!context_.account_internal_addr) {
+    error_ = "No internal address provided";
+    ScheduleWorkOnTask();
+    return;
+  }
+  context_.sync_state->AsyncCall(&OrchardSyncState::GetSpendableNotes)
+      .WithArgs(context_.account_id.Clone(),
+                context_.account_internal_addr.value())
+      .Then(base::BindOnce(
+          &ZCashCreateOrchardToTransparentTransactionTask::OnGetSpendableNotes,
+          weak_ptr_factory_.GetWeakPtr()));
+}
+
+void ZCashCreateOrchardToTransparentTransactionTask::OnGetSpendableNotes(
+    base::expected<std::optional<OrchardSyncState::SpendableNotesBundle>,
+                   OrchardStorage::Error> result) {
+  if (!result.has_value()) {
+    error_ = result.error().message;
+    ScheduleWorkOnTask();
+    return;
+  }
+
+  if (!result.value()) {
+    error_ = "No spendable notes";
+    ScheduleWorkOnTask();
+    return;
+  }
+
+  spendable_notes_ = std::move(*result.value());
+  ScheduleWorkOnTask();
+}
+
+void ZCashCreateOrchardToTransparentTransactionTask::CreateTransaction() {
+  CHECK(spendable_notes_);
+
+  ZCashTransaction zcash_transaction;
+
+  // Pick Orchard inputs.
+  auto pick_result =
+      PickZCashOrchardInputs(spendable_notes_->spendable_notes, amount_,
+                             ZCashTargetOutputType::kTransparent);
+  if (!pick_result) {
+    error_ = "Can't pick inputs";
+    ScheduleWorkOnTask();
+    return;
+  }
+
+  // Add Orchard inputs to transaction.
+  for (const auto& note : pick_result.value().inputs) {
+    OrchardInput orchard_input;
+    orchard_input.note = note;
+    zcash_transaction.orchard_part().inputs.push_back(std::move(orchard_input));
+  }
+  zcash_transaction.set_fee(pick_result->fee);
+
+  if (!context_.account_internal_addr) {
+    error_ = "Internal address error";
+    ScheduleWorkOnTask();
+    return;
+  }
+
+  if (!spendable_notes_->anchor_block_id) {
+    error_ = "Failed to select anchor";
+    ScheduleWorkOnTask();
+    return;
+  }
+
+  zcash_transaction.orchard_part().anchor_block_height =
+      spendable_notes_->anchor_block_id.value();
+
+  // Create transparent output for the recipient.
+  auto& transparent_output =
+      zcash_transaction.transparent_part().outputs.emplace_back();
+  transparent_output.address = transparent_address_;
+
+  // Calculate the amount to send.
+  uint64_t send_amount = amount_;
+  if (amount_ == kZCashFullAmount) {
+    // For max amount, send everything minups fees and change.
+    send_amount =
+        base::CheckSub<uint64_t>(zcash_transaction.TotalInputsAmount(),
+                                 zcash_transaction.fee(), pick_result->change)
```

Comment:
Are we supposed to subtract change for 'send all' case? There is no change expected.

Reply:
For the 'send all' case change just should be 0